### PR TITLE
Feature/ms 22/test uri added to create delivery event

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
https://oat-sa.atlassian.net/browse/MS-22

Payload request modification that is sent to manual scoring.
Added test_id for delivery.

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -29,7 +29,7 @@
   'label'       => 'Delivery Management',
   'description' => 'Manages deliveries using the ontology',
   'license'     => 'GPL-2.0',
-  'version'     => '10.1.0',
+  'version'     => '10.3.0',
```
Comment (peetya) [inline manifest.php]: Why not `10.2.0`?

File: model/DeliveryFactory.php [hunk 2]
```diff
@@ -90,7 +93,7 @@ public function create(core_kernel_classes_Class $deliveryClass, core_kernel_cla
             $propertyValues = $test->getPropertyValues($testPropretyInstance);
 
             if ($validationValue == 'notEmpty' && empty($propertyValues)) {
-                $report = \common_report_Report::createFailure(__('Test publishing failed because "%s" is empty.', $testPropretyInstance->getLabel()));
+                $report = \common_report_Report ::createFailure(__('Test publishing failed because "%s" is empty.', $testPropretyInstance->getLabel()));
```
Comment (peetya) [inline model/DeliveryFactory.php]: Please remove the space.

File: model/event/DeliveryCreatedEvent.php [hunk 3]
```diff
@@ -19,16 +19,24 @@
...
+     * @param string $deliveryUri
+     * @param string $testUri
+     *
+     * @throws \core_kernel_persistence_Exception
      */
-    public function __construct($deliveryUri)
+    public function __construct(core_kernel_classes_Resource $delivery)
     {
-        $this->deliveryUri = $deliveryUri;
+        $testProperty = new \core_kernel_classes_Property('http://www.tao.lu/Ontologies/TAODelivery.rdf#AssembledDeliveryOrigin');
```
Comment (siwane) [inline model/event/DeliveryCreatedEvent.php]: Could u use constant for this uri pls

File: model/event/DeliveryCreatedEvent.php [hunk 4]
```diff
@@ -19,16 +19,24 @@
...
+        $this->deliveryUri = $delivery->getUri();
+        $this->testUri = $delivery->getOnePropertyValue($testProperty)->getUri();
```
Comment (siwane) [inline model/event/DeliveryCreatedEvent.php]: I discourage to start process from constructor, e.g. access to database. Also it is possible to have a reference as delivery origin to a deleted test, could u create a test to ensure this point please

File: model/DeliveryFactory.php (hunk 2 above)
Comment (siwane) [reply model/DeliveryFactory.php]: +1

File: model/DeliveryFactory.php [hunk 5]
```diff
@@ -90,7 +93,7 @@ public function create(core_kernel_classes_Class $deliveryClass, core_kernel_cla
             $propertyValues = $test->getPropertyValues($testPropretyInstance);
 
             if ($validationValue == 'notEmpty' && empty($propertyValues)) {
-                $report = \common_report_Report::createFailure(__('Test publishing failed because "%s" is empty.', $testPropretyInstance->getLabel()));
+                $report = \common_report_Repor::createFailure(__('Test publishing failed because "%s" is empty.', $testPropretyInstance->getLabel()));
```
Comment (peetya) [inline model/DeliveryFactory.php]: typo: `Repor`

File: model/event/DeliveryCreatedEvent.php [hunk 6]
```diff
@@ -59,7 +69,7 @@ public function jsonSerialize()
      */
     public function getWebhookEventName()
     {
-        return 'DeliveryCreatedEvent';
+        return 'delivery-published';
```
Comment (peetya) [inline model/event/DeliveryCreatedEvent.php]: Based on our discussion, you need to revert this modification, right?

Comment (bartlomiejmarszal) [reply model/event/DeliveryCreatedEvent.php]: yes

File: model/event/DeliveryCreatedEvent.php [hunk 7]
```diff
@@ -17,18 +17,31 @@
...
+
 namespace oat\taoDeliveryRdf\model\event;
 
+use core_kernel_classes_Resource;
 use oat\tao\model\webhooks\WebhookSerializableEventInterface;
+use \core_kernel_persistence_Exception;
 
 class DeliveryCreatedEvent extends AbstractDeliveryEvent implements WebhookSerializableEventInterface
 {
+    const ASSEMBLED_DELIVERY_ORIGIN_URI = 'http://www.tao.lu/Ontologies/TAODelivery.rdf#AssembledDeliveryOrigin';
```
Comment (siwane) [inline model/event/DeliveryCreatedEvent.php]: This constant already exist: https://github.com/oat-sa/extension-tao-delivery-rdf/blob/develop/model/DeliveryAssemblyService.php#L40

**Pull request merged**